File: backend/app/schemas/football_club.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List, Dict

class FootballClubBase(BaseModel):
    name: str
    country: Optional[str] = None
    league: Optional[str] = None
    founded_year: Optional[int] = None
    stadium: Optional[str] = None
    capacity: Optional[int] = None
    manager: Optional[str] = None
    nickname: Optional[str] = None
    image_url: Optional[str] = None
    website: Optional[str] = None
    description: Optional[str] = None
    ranking: Optional[int] = None
    
    # Enhanced Statistics
    total_trophies: Optional[int] = 0
    market_value: Optional[str] = None
    league_position: Optional[int] = None
    domestic_ranking: Optional[int] = None
    captain: Optional[str] = None
    owner: Optional[str] = None
    main_rivals: Optional[str] = None
    average_attendance: Optional[int] = None
    
    # Detailed Honors
    honors_json: Optional[Dict[str, int]] = None

class FootballClubCreate(FootballClubBase):
    pass

class FootballClubUpdate(FootballClubBase):
    name: Optional[str] = None

class FootballClub(FootballClubBase):
    id: int
    last_updated: datetime

    class Config:
        from_attributes = True

class FootballClubList(BaseModel):
    items: List[FootballClub]
    total: int
    page: int
    size: int

File: frontend/lib/models/football_club.dart
class FootballClub {
  final int id;
  final String name;
  final String? country;
  final String? league;
  final int? foundedYear;
  final String? stadium;
  final int? capacity;
  final String? manager;
  final String? nickname;
  final String? imageUrl;
  final String? website;
  final String? description;
  final int? ranking;
  
  // Enhanced Statistics
  final int totalTrophies;
  final String? marketValue;
  final int? leaguePosition;
  final String? captain;
  final String? owner;
  final String? mainRivals;
  final int? averageAttendance;

  FootballClub({
    required this.id,
    required this.name,
    this.country,
    this.league,
    this.foundedYear,
    this.stadium,
    this.capacity,
    this.manager,
    this.nickname,
    this.imageUrl,
    this.website,
    this.description,
    this.ranking,
    this.totalTrophies = 0,
    this.marketValue,
    this.leaguePosition,
    this.captain,
    this.owner,
    this.mainRivals,
    this.averageAttendance,
  });

  factory FootballClub.fromJson(Map<String, dynamic> json) {
    return FootballClub(
      id: json['id'],
      name: json['name'],
      country: json['country'],
      league: json['league'],
      foundedYear: json['founded_year'],
      stadium: json['stadium'],
      capacity: json['capacity'],
      manager: json['manager'],
      nickname: json['nickname'],
      imageUrl: json['image_url'],
      website: json['website'],
      description: json['description'],
      ranking: json['ranking'],
      totalTrophies: json['total_trophies'] ?? 0,
      marketValue: json['market_value'],
      leaguePosition: json['league_position'],
      captain: json['captain'],
      owner: json['owner'],
      mainRivals: json['main_rivals'],
      averageAttendance: json['average_attendance'],
    );
  }
}

class FootballClubListResponse {
  final List<FootballClub> items;
  final int total;
  final int page;
  final int size;

  FootballClubListResponse({
    required this.items,
    required this.total,
    required this.page,
    required this.size,
  });

  factory FootballClubListResponse.fromJson(Map<String, dynamic> json) {
    return FootballClubListResponse(
      items: (json['items'] as List)
          .map((i) => FootballClub.fromJson(i))
          .toList(),
      total: json['total'],
      page: json['page'],
      size: json['size'],
    );
  }
}
